### PR TITLE
Feature/qr for browser

### LINKED ISSUE #12194: Add QR code scanning ability to Status browser
As a user, I want to be able to scan a URL from Status browser instead of typing it. 

note: this URL should either be a general URL (e.g www.cryptokitties.co) or a join.status.im/xxx URL

### LINKED ISSUE #12195: Redirect to browser from wallet QR scanner
As a user, when I am using the wallet scanner on the main wallet screen (see below) and I am scanning an URL, Status should open status browser. 

notes: 
- current behaviour: it shows an error invalid address
- when the QR code scanner is used to fill up the to field of a transaction in send/choose recipient , and a URL is scanned, it should not redirect to the browser. 

IMG_A0D233717BC5-1

### LINKED ISSUE #12222: chat QR code scanner should be able to open a URL in the browser
IN the 'chat' section' under the '+' button there is a QR code scanner
This scanner should be universal: when scanning a URL it should open this URL in Status browser
note: this URL should either be joins.status.im/xxx or a simple UR (e.g www.cryptokitties.co)
current behaviour: error message "unable to read this code"

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #12194
fixes #12222
fixes #12195

only last commit for review, this PR is dependent on rnn leftovers PR

### CODE REVIEW

File: src/status_im/navigation/core.cljs [hunk 1]
```diff
@@ -116,6 +118,7 @@
          (re-frame/dispatch [:set :view-id @pushed-screen-id])))
 
      (let [comp @dissmissing]
+       (println "DISMISSING" comp)
```
Comment (shivekkhurana) [inline src/status_im/navigation/core.cljs]: Rogue `println`

**Review by shivekkhurana — APPROVED** [review]

**Comment by qoqobolo** [comment]
@flexsurfer thank you for the PR!

#### ISSUE 1: `Open in Status / in OS browser` menu opens twice if scan `joins.status.im/xxx` link with scanner in the Wallet view

**_Steps:_**
1. Open Wallet
2. Open QR scanner
3. Scan link e.g. https://join.status.im/b/https://www.larvalabs.com/cryptopunks

**_OS:_** iOS, Android

https://user-images.githubusercontent.com/67952253/122744585-6792ef00-d288-11eb-9973-8ea3d3baf027.mp4

**Comment by qoqobolo** [comment]
#### ISSUE 2: When scanning URL with the scanner in the `Recipient` screen, it is still asked to redirect to the Status browser.

Not sure if this is expected behavior according to one of the notes in https://github.com/status-im/status-react/issues/12195:
> when the QR code scanner is used to fill up the to field of a transaction in send/choose recipient, and a URL is scanned, it should not redirect to the browser.

**_Steps:_**
1. Open Wallet
2. Open `Send transaction` dialog
3. Tap `Choose Recipient`
4. Open QR scanner
5. Scan some link, e.g. https://join.status.im/b/https://www.larvalabs.com/cryptopunks

**Expected result:** invalid address error
**Actual result:** redirect to browser

**Comment by guylouis** [comment]
thanks @qoqobolo and @flexsurfer, some notes
*issue 1* I reproduce it too 
*issue 2* indeed it would be better if the QR scanner only opens the browser in the wallet section on the QR code scanner form the wallet page (as it deos with this PR) and not from the scanner when the user is engaged in sending a transaction. copying @hesterbruikman @errorists here, since I don't have a super strong opinion about it either, and I think we could merge without issue 2 solved.

**Comment by flexsurfer** [comment]
thanks @qoqobolo all fixed

**Comment by hesterbruikman** [comment]
_@errorists below for your review_

To summarize, IMO these are the UI requirements: 

1. Scan url from chat bottom sheet, wallet overview, browser (bottom sheet) > Show prompt to Open in Status or Android/iOS
2. Scan url from address input (wallet, chat, enode) > Skip prompt and place address
   

3. Position in browser > Show on Bottom sheet in the same way as QR scan is shown on the chat bottom sheet, including header `Open...` on the left. (For consistency)
          - One issue I see is that currently the `...` in browser is not available on `Empty tabs`. This makes sense as most items are not useful (no site to Share, Connect to, add as favorite. I would enable a bottom sheet on Empty tabs with only the New tab and scan item

I like the idea of having the QR code scanning next to the address bar. However, I think it needs more effort to design the transition to show it on empty tabs and hide it on page load.

**Comment by errorists** [comment]
I'll copy-paste what I answered in core, `the original idea way back was to add it above keyboard similar to mobile Chrome. imo, placing it in the toolbar is preferable, so in the tabs overview you would replace the greyed out ••• with QR button and once inside a browser tab when there are more icons in the toolbar to display and they don't all fit, have them placed in the ••• overflow with the qr code button.`

I'm _strongly not_ in favour of placing it in the Top Bar, it's a constrained space and we made significant effort to clean it up not long ago before.

**Comment by qoqobolo** [comment]
@flexsurfer thanks for the fixes! Works great, no new issues were found.
If no further changes are planned, PR can be merged.

**Pull request merged**